client/src/Components/Register/RegisterForm/RegisterForm.js

```diff
@@ -1,38 +1,125 @@
-import React from "react";
+import React, { useState } from "react";
+import axios from "axios";
 import "./RegisterForm.css";
+import { useForm } from "react-hook-form";
+import { RHFInput } from "react-hook-form-input";
 import { TextField, Button } from "@material-ui/core";
+import { Alert } from "@material-ui/lab";
+import { makeStyles } from "@material-ui/core/styles";
+import { Redirect } from "react-router-dom";
 
-class RegisterForm extends React.Component {
-  state = {};
-  render() {
-    return (
-      <form>
-        <div class="flex-container">
-          <TextField
-            id="email"
-            label="Your Email"
-            type="email"
-            variant="outlined"
-          ></TextField>
-          <TextField
-            id="name"
-            label="Your name"
-            type="string"
-            variant="outlined"
-          ></TextField>
-          <TextField
-            id="email"
-            label="Create Password"
-            type="string"
-            variant="outlined"
-          ></TextField>
-        </div>
-        <Button variant="contained" color="red">
+const useStyles = makeStyles({
+  input: {
+    marginTop: `5px`,
+    marginBottom: `5px`,
+  },
+  button: {
+    margin: `5px`,
+    maxWidth: `200px `,
+  },
+});
+
+function RegisterForm() {
+  const classes = useStyles();
+  const [registerError, setRegisterError] = useState(false);
+  const [errMsg, setErrMsg] = useState("");
+  const { register, handleSubmit, setValue, errors } = useForm();
+  const [isRegistered, setIsRegistered] = useState(false);
+
+  const onSubmit = async (data) => {
+    const payload = {
+      email: data.email,
+      password: data.password,
+    };
+    try {
+      const createUserRes = await axios.post(
+        "http://localhost:3001/register",
+        payload
+      );
+      const createProfileRes = await axios.post(
+        "http://localhost:3001/profile",
```

Comment:
Should not be sending the requests to localhost. It should be domain agnostic.
eg. `await axios.post('/register')`

Reply:
thanks made the fix.